src/views/login/EnrollmentView.tsx

```diff
@@ -26,42 +22,34 @@ export function EnrollmentView() {
     callsign ?? ""
   }&&approvalcode=${approveCode ?? ""}`;
 
+  // Redirect to login if approveCode or callsign is missing
   useEffect(() => {
     if (!approveCode || !callsign) {
       navigate("/login");
     }
   }, [approveCode, callsign, navigate]);
 
   const [isEnrolled, setIsEnrolled] = useState(false);
+
+  // Check enrollment status periodically
   useOwnEnrollmentStatus({
-    onSuccess: (isEnrolled) => {
-      if (isEnrolled) {
+    onSuccess: (enrolled) => {
+      if (enrolled) {
         setIsEnrolled(true);
       }
     },
     refetchInterval: 1000,
     enabled: !isEnrolled,
   });
 
-  if (isEnrolled) {
-    return (
-      <Layout showNavbar={true} showFooter={true}>
-        <main className="px-10 flex flex-col gap-3 items-center justify-start h-full">
-          <Text title={subdomain || "Loading..."} />
-          <span className="text-white">{t("enrollment-approved")}</span>
-          <Button
-            onClick={() => {
-              navigate("/login/createmtls");
-              window.location.reload();
-            }}
-          >
-            {t("continue-clicking-here")}
-          </Button>
-        </main>
-      </Layout>
-    );
-  }
+  // Redirect to createmtls when enrollment is approved
+  useEffect(() => {
+    if (isEnrolled) {
+      navigate("/login/createmtls");
+    }
```

Comment:
Why not navigate directly at `onSuccess` above?

Reply:
true that, unnecessary -> fix'd